source/MRCuda/MRCudaSolarRadiation.cu

```diff
@@ -0,0 +1,98 @@
+#include "MRCudaSolarRadiation.cuh"
+#include "MRCudaBasic.h"
+#include "device_launch_parameters.h"
+#include "MRCudaFloat.cuh"
+#include <float.h>
+
+namespace MR
+{
+namespace Cuda
+{
+
+__global__ void rayKernel( const Node3* nodes, const float3* meshPoints, const FaceToThreeVerts* faces, const float3* samples, const uint64_t* validSamples, const IntersectionPrecomputes* precs, uint64_t* res, const size_t resBlockCount, const size_t sampleCount, const size_t precCount )
+{
+    if ( resBlockCount == 0 )
+    {
+        assert( false );
+        return;
+    }
+
+    const size_t blockIndex = blockIdx.x * blockDim.x + threadIdx.x;
+    if ( blockIndex >= resBlockCount )
+        return;
+
+    const size_t blockStart = ( blockIndex << 6 );
```

Comment:
why 6?

Reply:
There are 64 bits in each block. It is faster alternative to multiplying on 64